### DISCUSSION TITLE
How to inspect server launch failure?

### DISCUSSION THREAD

**Discussion by gabyx** [mfussenegger/nvim-dap#1115] [General]
I have `dlv` in my path but I am unable to launch the debug in a go file. I am on NixOS and `dlv` is in the path and works.

All I get is `DAP  ERROR Error on launch: Failed to launch`. How can I get more insight why the server could not get started. I followed the simple Go setup which did not work.
Also the `nvim-dap-go` extension has the same problem so I expect that its a local problem?

Any help on getting a better log output? Thanks a lot!!

**Comment by keystroke3** [comment]
Hey, if you are still having this issue, I found that you can use toggle [dabui](https://github.com/rcarriga/nvim-dap-ui), and it will show you in the bottom ~~left~~ right corner what is happening.

**Comment by MondayHopscotch** [comment]
I'm able to launch a standard debug just fine, but I get this error when trying to debug a test via the documented `:lua require('dap-go').debug_test()`. It just outputs:
```
starting debug session './. : TestHello'...
Error on launch: Failed to launch
```

Logs don't show much:
```
[ INFO ] 2024-01-21T15:45:12Z-0700 ] ...nvim/site/pack/packer/start/nvim-dap/lua/dap/session.lua:1788 ]	"Session closed due to disconnect"
```
So I'm not sure what the issue is yet.

**Comment by MaMrEzO** [comment]
Set log level to trace:
`:DapSetLogLevel TRACE`
try to debug, and then check logs again.
